var/spack/repos/builtin/packages/py-cartopy/package.py

```diff
@@ -21,7 +22,8 @@ class PyCartopy(PythonPackage):
     depends_on('py-pyshp@1.1.4:',   type=('build', 'run'))
     depends_on('py-six@1.3.0:',     type=('build', 'run'))
     depends_on('geos@3.3.3:')
-    depends_on('proj@4.9.0:5')
+    depends_on('proj@4.9.0:5', when='@0.16.0')
+    depends_on('proj@6:',      when='@0.17.0')
```

Comment:
Are you sure PROJ 4 and 5 don't work with 0.17.0? The `setup.py` still lists 4.9.0 as the minimum PROJ version.

Reply:
No. I just wanted to keep the recipe backwards compatible.